### ISSUE TITLE
Let `require-jsdoc` rule not need inherited classes to declare documentation for overridden accessors and constructors.

### ISSUE THREAD

**Issue by Sven65** [eslint/eslint#14200]
**What rule do you want to change?**

[`require-jsdoc`](https://eslint.org/docs/rules/require-jsdoc)

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**

I propose two new options, as accessors, and the constructor are different concepts for an ES6 class.

The first one would let definitions of overridden **accessors** be allowed, with a new JSDoc comment being optional.

The second one would let overridden **constructors** be allowed, , with a new JSDoc comment being optional.

**Please provide some example code that this change will affect:**

```js
class MyClass {
	/**
	 * Constructs a new class
	 * @param {string} id The ID of the class
	 */
	constructor (id) {
		this._id = id
	}

	/**
	 * The ID of the class
	 * @type {string}
	 */
	get id () {
		return this._id
	}

	/**
	 * Get a random string from the class
	 * @type {?string}
	 */
	get randomString () {
		return null
	}

	/**
	 * Checks if the id is the same
	 * @param {string} id The ID to check
	 * @function
	 * @returns {boolean}
	 */
	isIdSame (id) {
		return this.id === id
	}
}

class ExtendedClass extends MyClass {
	constructor () {
		super("1")
	}

	get randomString () {
		const items = [
			"String 1",
			"String 2",
			"String 3"
		]

		return items[Math.floor(Math.random() * items.length)]
	}
}
```

**What does the rule currently do for this code?**

`ExtendedClass` spews errors, as it doesn't have its accessors, or constructor documented, even though its parent does.

**What will the rule do after it's changed?**

`ExtendedClass` will not spew errors, as the parent class has its accessors and constructor documented.

**Are you willing to submit a pull request to implement this change?**

I would be able to, but I'm not sure that this change is entirely possible to do, nor am I very well versed in the eslint API.

**Comment by aladdin-add** [comment]
thanks for filing the issue, while the rule has been deprecated - we do not make any change to deprecated rules.

meanwhile, I suggest to take a look at the [eslint-plugin-jsdoc](https://www.npmjs.com/package/eslint-plugin-jsdoc).

**Comment by Sven65** [comment]
Thanks! I'll give it a look, I'll reopen the issue if the plugin you linked doesn't satisfy my needs.
